### PR TITLE
Revised API documentation theme

### PR DESCRIPTION
**Pull request by szadam**
Revised API documentation theme to be more similar to UR.
Preview:
https://szadam.github.io/unified-memory-framework/

### CODE REVIEW

File: third_party/requirements.txt [hunk 1]
```diff
@@ -3,6 +3,5 @@
 clang-format==15.0.7
 # Generating HTML documentation
 breathe==4.35.0
-docutils==0.19
```
Comment (PatKamin) [inline third_party/requirements.txt:6]: `docutils` maximum version is `0.17` for sphinx `4.5`

**Comment by PatKamin** [comment]
Please, merge commits.

File: third_party/requirements.txt (hunk 1 above)
Comment (PatKamin) [reply third_party/requirements.txt:6]: Discussed offline

**Review by PatKamin — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]
LGTM

**Pull request merged**